redisson/src/main/java/org/redisson/client/protocol/decoder/StreamInfoDecoder.java

```diff
@@ -38,29 +38,70 @@
 
     @Override
     public StreamInfo<Object, Object> decode(List<Object> parts, State state) {
+        StreamInfoWrapper fields = new StreamInfoWrapper(parts);
```

Comment:
We don't need any wrapper here. List could be converted to map:
```java
        Map<String, Object> map = IntStream.range(0, parts.size()).boxed().collect(
                Collectors.groupingBy(e -> e / 2, Collectors.mapping(e -> parts.get(e), Collectors.toList())))
                .values().stream().collect(Collectors.toMap(e -> e.get(0), e -> e.get(1)));
```

Reply:
I added the wrapper for mental support only, but this change is a very interesting solution.

Reply:
I added the requested changes to the proposed solution.